qiskit/backends/aer/qasm_simulator.py

```diff
@@ -109,11 +110,13 @@ def run(self, qobj):
 
     def _run_job(self, job_id, qobj):
         self._validate(qobj)
-        result = run(qobj, self._configuration.exe)
+        if isinstance(qobj, Qobj):
+            qobj_dict = qobj.as_dict()
+        else:
+            qobj_dict = qobj
```

Comment:
Can't we always assume that a Qobj object is passed here, rather than dict? I have assumed this in the python simulators. @diego-plan9 can decide, I'm fine with either case.

Reply:
lets make it qobj

Reply:
Yes, it should always receive a `qobj` - I remember complaining about the same for a similar change in the clifford! I'll adjust it.